.github/workflows/pycodestyle.yml

```
@@ -0,0 +1,22 @@
+name: pycodestyle
+
+on: [push]
```

The rule in your other workflow is nicer:

```suggestion
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
```